src/arrayMethodJoin.js

```diff
@@ -4,8 +4,17 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    for (let i = 0; i <= this.length; i++) {
+      const isLastItem = i < this.length - 1;
+      const value = this[i] !== undefined && this[i] !== null ? this[i] : '';
```

Comment:
I guess it's better create variable for our condition (just for ternary operator to look smaller) + variable for this[i]